test/perf/benchmarks/README.md

```diff
@@ -0,0 +1,88 @@
+## Micro Benchmarks
+
+This folder contains micro benchmarks that test the performance of PowerShell Engine.
+
+### Requirement
+
+1. A good suite of benchmarks
+   Something that measures only the thing that we are interested in and _produces accurate, stable and repeatable results_.
+2. A set of machine with the same configurations.
+3. Automation for regression detection.
+
+### Design Decision
+
+1. This project is internal visible to `System.Management.Automation`.
+   We want to be able to target some internal APIs to get measurements on specific scoped scenarios,
+   such as measuring the time to compile AST to a delegate by the compiler.
+2. This project makes `ProjectReference` to other PowerShell assemblies.
+   This makes it easy to run benchmarks with the changes made in the codebase.
+   To run benchmarks with a specific version of PowerShell,
+   just replace the `ProjectReference` with a `PackageReference` to the `Microsoft.PowerShell.SDK` NuGet package of the corresponding version.
+
+### Quick Start
+
+You can run the benchmarks directly using `dotnet run` in this directory:
+1. To run the benchmarks in Interactive Mode, where you will be asked which benchmark(s) to run:
+   ```
+   dotnet run -c release
+   ```
+
+2. To list all available benchmarks ([read more](https://github.com/dotnet/performance/blob/main/docs/benchmarkdotnet.md#Listing-the-Benchmarks)):
+   ```
+   dotnet run -c release --list [flat/tree]
+   ```
+
+3. To filter the benchmarks using a glob pattern applied to `namespace.typeName.methodName` ([read more](https://github.com/dotnet/performance/blob/main/docs/benchmarkdotnet.md#Filtering-the-Benchmarks)]):
+   ```
+   dotnet run -c Release -f net6.0 --filter *script* --list flat
+   ```
+
+4. To profile the benchmarked code and produce an ETW Trace file ([read more](https://github.com/dotnet/performance/blob/main/docs/benchmarkdotnet.md#Profiling))
+   ```
+   dotnet run -c Release -f net6.0 --filter *script* --profiler ETW
+   ```
+
+You can also use the function `Start-Benchmarking` from the module [`perf.psm1`](../perf.psm1) to run the benchmarks:
+```powershell
+Start-Benchmarking [[-TargetPSVersion] <string>] [[-List] <string>] [[-Filter] <string[]>] [[-Artifacts] <string>] [-KeepFiles] [<CommonParameters>]
+```
+Run `Get-Help Start-Benchmarking -Full` to see the description of each parameter.
+
+### Regression Detection
```

Comment:
Thanks for adding full example!

Looking on the scenario I guess it is mainstream scenario and it would be great to run it with one command.

Reply:
That can be easily done on top of the existing functions. I believe it will be what we need for the perf lab infra, but we want to leverage .NET team's existing perf lab infra if possible.